## Long read genome assembly

#### Basecalling using `guppy`
We used `guppy v0.15.3` within the MINKNOW software for basecalling nanopore runs. 

#### Assembly with `Canu`
I ran `Canu v2.2` as follows:

```ShellSession
# Assembly Arachnopeziza aurata
canu -java=$HOME/miniconda3/pkgs/java-jdk-8.0.92-1/bin/java \
gridEngineResourceOption='--cpus-per-task=THREADS -–mem-per-cpu=MEMORY' \
useGrid=false maxMemory=128G maxThreads=8 \
-p Aaurata_canu -d $READ_DIR/Aaurata_nanopore \
genomeSize=42m \
-nanopore-raw SRR30482969.fastq.gz

# Assembly Arachnopeziza aurelia
canu -java=$HOME/miniconda3/pkgs/java-jdk-8.0.92-1/bin/java \
gridEngineResourceOption='--cpus-per-task=THREADS -–mem-per-cpu=MEMORY' \
useGrid=false maxMemory=128G maxThreads=8 \
-p Aaurelia_canu -d $READ_DIR/Aaurelia_nanopore \
genomeSize=42m \
-nanopore-raw SRR30482968.fastq.gz
```

#### Assembly with `Flye`
I used `Flye v2.9.2` as assembler as follows:

```ShellSession
# Assembly Arachnopeziza aurata
flye --nano-hq $READ_DIR/SRR30482969.fastq.gz \
--out-dir $HOME/Aaurata_flye \
--iterations 3 --threads 12 \
--genome-size 42m --asm-coverage 50 -m 10000

# Assembly Arachnopeziza aurelia
flye --nano-hq $READ_DIR/SRR30482968.fastq.gz \
--out-dir $HOME/Aaurelia_flye \
--iterations 3 --threads 12 \
--genome-size 42m --asm-coverage 50 -m 10000
```

#### Assembly with `NextDenovo`
I used `NextDenovo v2.5.2` as assembler. `NextDenovo` requires a config-file to run `config/Aaurata_run.cfg`, which looked as below. The file $HOME/Aaurata_nd/config/Aaurata.fofn contains the location of the read input file, $READ_DIR/SRR30482969.fastq.gz for *A. aurata*, and $HOME/Aaurelia_nd/config/Aaurelia.fofn contains the path $READ_DIR/SRR30482968.fastq.gz for *A. aurelia*.
```
[General]
job_type = local # here we use SGE to manage jobs
job_prefix = Aaurata
task = all
rewrite = yes
rerun = 3
deltmp = yes
parallel_jobs = 12
input_type = raw
read_type = ont # clr, ont, hifi
input_fofn = $HOME/Aaurata_nd/config/Aaurata.fofn
workdir = $HOME/Aaurata_nd

[correct_option]
read_cutoff = 1k
genome_size = 40m # estimated genome size
sort_options = -m 10g -t 8
minimap2_options_raw = -t 8
pa_correction = 8
correction_options = -p 10

[assemble_option]
minimap2_options_cns = -t 8
minimap2_options_map = -t 8
nextgraph_options = -a 1 -q 10 -E 5000
```

Then, I ran the assembler: 
```ShellSession
# Assembly Arachnopeziza aurata
nextDenovo $HOME/Aaurata_nd/config/Aaurata_run.cfg

# Assembly Arachnopeziza aurelia
nextDenovo $HOME/Aaurelia_nd/config/Aaurelia_run.cfg
``` 

#### Consolidating the assemblies with `quickmerge`
I used `quickmerge v0.3` to merge the assemblies obtained with the three assembly strategies. 

```ShellSession
# Assembly Arachnopeziza aurata
merge_wrapper.py \
$HOME/Aaurata_canu/Aaurata_canu.fasta \
$HOME/Aaurata_flye/assembly.fasta \
$HOME/Aaurata_nd/03.ctg_graph/nd.asm.fasta \
-pre $HOME/Aaurata_merged.fasta

# Assembly Arachnopeziza aurelia
merge_wrapper.py \
$HOME/Aaurelia_canu/Aaurelia_canu.fasta \
$HOME/Aaurelia_flye/assembly.fasta \
$HOME/Aaurelia_nd/03.ctg_graph/nd.asm.fasta \
-pre $HOME/Aaurelia_merged.fasta
```

#### Polishing assembly with `pilon`
To polish the genome assemblies, I first remapped short-read whole genome sequencing data with `BWA v0.7.17-r1188`:

```ShellSession
# Assembly Arachnopeziza aurata
bwa index $HOME/Aaurata_merged.fasta # create index
# map short reads
bwa mem -t 8 -M $HOME/Aaurata_merged.fasta \
$READ_DIR/SRR30482971_R1_trim.fq.gz $READ_DIR/SRR30482971_R2_trim.fq.gz \
-o $HOME/Aaurata_merged.bwa_NGS.sam
# parse data
samtools sort -l 0 -o $HOME/Aaurata_merged.bwa_NGS.bam -O bam $HOME/Aaurata_merged.bwa_NGS.sam | # sort
samtools view -F 0x04 -b - > $HOME/Aaurata_merged.bwa_NGS.bam # remove unmapped reads
samtools index -b $HOME/Aaurata_merged.bwa_NGS.bam
```

```ShellSession
# Assembly Arachnopeziza aurelia
bwa index $HOME/Aaurelia_merged.fasta # create index
# map short reads
bwa mem -t 8 -M $HOME/Aaurelia_merged.fasta \
$READ_DIR/SRR30482970_R1_trim.fq.gz $READ_DIR/SRR30482970_R2_trim.fq.gz \
-o $HOME/Aaurelia_merged.bwa_NGS.sam
# parse data
samtools sort -l 0 -o $HOME/Aaurelia_merged.bwa_NGS.bam -O bam $HOME/Aaurelia_merged.bwa_NGS.sam | # sort
samtools view -F 0x04 -b - > $HOME/Aaurelia_merged.bwa_NGS.bam # remove unmapped reads
samtools index -b $HOME/Aaurelia_merged.bwa_NGS.bam
```

Then, `pilon v1.24` was used to polish the assemblies like this:

```ShellSession
# Assembly Arachnopeziza aurata
pilon -Xmx12G \
--genome $HOME/Aaurata_merged.fasta \
--frags $HOME/Aaurata_merged.bwa_NGS.bam \
--output $HOME/Aaurata_merged.pilon.fa

# Assembly Arachnopeziza aurelia
pilon -Xmx12G \
--genome $HOME/Aaurelia_merged.fasta \
--frags $HOME/Aaurelia_merged.bwa_NGS.bam \
--output $HOME/Aaurelia_merged.pilon.fa
```


#### Discovering and extending telomeric ends with `teloclip`

Telomeres were manually identified at the ends of assembled contigs as telomeric repeats 5’-TTAGGG-3’ or 3’-CCCTAA-5’. To complete the sequence ends where telomeric repeats were not found, we used `teloclip v0.0.4` [Manual](https://github.com/Adamtaranto/teloclip). First, the long reads were mapped to the genome assemblies using `Minimap2 v2.26-r1175` ([Manual](https://github.com/lh3/minimap2)):

```ShellSession
# Assembly Arachnopeziza aurata
minimap2 -t 12 -k 20 -ax map-ont $HOME/Aaurata_merged.pilon.fa \
$READ_DIR/SRR30482969.fastq.gz | \
samtools view -h -F 0x2308 - | samtools sort > ${WORK_DIR}/Aaurata_merged.minimap.bam

# Assembly Arachnopeziza aurelia
minimap2 -t 12 -k 20 -ax map-ont $HOME/Aaurelia_merged.pilon.fa \
$READ_DIR/SRR30482968.fastq.gz | \
samtools view -h -F 0x2308 - | samtools sort > ${WORK_DIR}/Aaurelia_merged.minimap.bam
```

Next, `SAMtools v1.8` ([]()) in conjunction with `teloclip` were used to filter reads mapping at contig ends. 
```ShellSession
# Assembly Arachnopeziza aurata

# clip reads
samtools view -h ${WORK_DIR}/Aaurata_merged.minimap.bam | \
teloclip --refIdx $HOME/Aaurata_merged.pilon.fa.fai | \
samtools sort > ${WORK_DIR}/Aaurata_merged.minimap.clipped.bam
# filter reads
samtools view -h ${WORK_DIR}/Aaurata_merged.minimap.clipped.bam | \
teloclip --refIdx $HOME/Aaurata_merged.pilon.fa.fai --motifs TTAGGG,TTAAGGG --matchAny | \
samtools sort > ${WORK_DIR}/Aaurata_merged.minimap.clipped_filtered.bam
# extract reads
samtools view -h ${WORK_DIR}/Aaurata_merged.minimap.clipped_filtered.bam | \
teloclip --ref $HOME/Aaurata_merged.pilon.fa.fai --motifs TTAGGG,TTAAGGG | \
teloclip-extract --refIdx $HOME/Aaurata_merged.pilon.fa.fai --extractReads --extractDir SplitOverhang
```

```ShellSession
# Assembly Arachnopeziza aurelia

# clip reads
samtools view -h ${WORK_DIR}/Aaurelia_merged.minimap.bam | \
teloclip --refIdx $HOME/Aaurelia_merged.pilon.fa.fai | \
samtools sort > ${WORK_DIR}/Aaurelia_merged.minimap.clipped.bam
# filter reads
samtools view -h ${WORK_DIR}/Aaurelia_merged.minimap.clipped.bam | \
teloclip --refIdx $HOME/Aaurelia_merged.pilon.fa.fai --motifs TTAGGG,TTAAGGG --matchAny | \
samtools sort > ${WORK_DIR}/Aaurelia_merged.minimap.clipped_filtered.bam
# extract reads
samtools view -h ${WORK_DIR}/Aaurelia_merged.minimap.clipped_filtered.bam | \
teloclip --ref $HOME/Aaurelia_merged.pilon.fa.fai --motifs TTAGGG,TTAAGGG | \
teloclip-extract --refIdx $HOME/Aaurelia_merged.pilon.fa.fai --extractReads --extractDir SplitOverhang
```

Then, the reads of each chromosome end were aligned via multiple sequence alignment using `MAFFT v7.520`. The alignments could then be used to manually extend scaffolds, yielding the genome assembly files `Aaurata_merged.teloclip.fasta` and `Aaurelia_merged.teloclip.fasta`. 
```ShellSession
for file in ${WORK_DIR}/SplitOverhang/*.fasta; do
base=${file%.fasta}
awk '/^[^>]/ { gsub(/[A-Z]/, ""); if (length > 0) print } /^[>]/ { print }' $file | mafft --auto - > $base.mafft.fasta;
done
```
