## Long read genome assembly

#### Basecalling using `guppy`
We used `guppy v0.15.3` within the MINKNOW software for basecalling nanopore runs. 

#### Assembly with `Canu`
I ran `Canu v2.2` as follows:

```ShellSession
spades.py -1 S1529Nr1_R1_trim.corrected.fq.gz -2 S1529Nr1_R2_trim.corrected.fq.gz -s S1529Nr1_unpaired_trim.corrected.fq.gz --only-assembler -k 31,51,71,91,121 -o Ltaurica_SPAdes
```

#### Assembly with `Flye`
I used `Flye v2.9.2` as assembler as follows:

```ShellSession
flye --iterations 3 --threads 12 --genome-size 42m --asm-coverage 50 -m 10000
```

#### Assembly with `NextDenovo`
I used `NextDenovo v2.5.2` as assembler. `NextDenovo` requires a config-file to run, which looked like this: 
```
#maximal read length
max_rd_len=150
[LIB]
#average insert size
avg_ins=300
#if sequence needs to be reversed 
reverse_seq=0
#in which part(s) the reads are used
asm_flags=3
#use only first 100 bps of each read
rd_len_cutoff=150
#in which order the reads are used while scaffolding
rank=1
# cutoff of pair number for a reliable connection (at least 3 for short insert size)
pair_num_cutoff=3
#minimum aligned length to contigs for a reliable read location (at least 32 for short insert size)
map_len=32
#a pair of fastq file, read 1 file should always be followed by read 2 file
q1=/hpcwork/sk893857/DNA-Seq/trimmed/S1529Nr1_R1_trim.corrected.fq.gz
q2=/hpcwork/sk893857/DNA-Seq/trimmed/S1529Nr1_R2_trim.corrected.fq.gz
#fastq file for single reads
q=/hpcwork/sk893857/DNA-Seq/trimmed/S1529Nr1_unpaired_trim.corrected.fq.gz
```
Then, I ran the assembler: 
```ShellSession
SOAPdenovo-127mer all -p 48 -s config-file -K 127 -R -N 190000000 -o Ltaurica_SOAP127 1>Ltaurica_assembly.log 2>Ltaurica_assembly.err
```
`SOAPdenovo2` offers two binaries (-127mer and -63mer) optimized for different maximum kmers. `-R` means resolve repeats by reads. `-N 190000000` indicates the approximate genome size. `-K 127` sets the kmer to 127. 

#### Consolidating the assemblies with `RagTag`
I used `RagTag v2.1.0` to merge the assemblies obtained with the three assembly strategies. 

```ShellSession
SOAPdenovo-127mer all -p 48 -s config-file -K 127 -R -N 190000000 -o Ltaurica_SOAP127 1>Ltaurica_assembly.log 2>Ltaurica_assembly.err
```

#### Polishing assembly with `pilon`
To polish the genome assemblies, I first remapped short-read whole genome sequencing data with `BWA v0.7.17-r1188`:

```ShellSession
SOAPdenovo-127mer all -p 48 -s config-file -K 127 -R -N 190000000 -o Ltaurica_SOAP127 1>Ltaurica_assembly.log 2>Ltaurica_assembly.err
```

Then, `pilon v1.24` was used to polish the assemblies like this:

```ShellSession
SOAPdenovo-127mer all -p 48 -s config-file -K 127 -R -N 190000000 -o Ltaurica_SOAP127 1>Ltaurica_assembly.log 2>Ltaurica_assembly.err
```
