#### Using `QUAST` to obtain assembly statistics and compare assemblies
I used `Quast v5.2.0` to generate various assembly statistics (see [Manual](https://quast.sourceforge.net/docs/manual.html) for details). 
```ShellSession
# Assembly Arachnopeziza aurata
python $HOME/miniconda3/envs/assembly/bin/quast.py \
-t 12 --report-all-metrics \
-b --fungus --k-mer-stats \
--nanopore $READ_DIR/SRR30482969.fastq.gz \
--pe1 $READ_DIR/SRR30482971_R1_trim.fq.gz \
--pe2 $READ_DIR/SRR30482971_R2_trim.fq.gz \
--labels "Aaurata_Canu, Aaurata_Flye, Aaurata_NextDenovo, Aaurata_quickmerge" \
-o ${WORK_DIR}/quast \
$HOME/Aaurata_canu/Aaurata_canu.fasta \
$HOME/Aaurata_flye/assembly.fasta \
$HOME/Aaurata_nd/03.ctg_graph/nd.asm.fasta \
$HOME/Aaurata_merged.fasta \
$HOME/Aaurata_merged.teloclip.fasta

# Assembly Arachnopeziza aurelia
python $HOME/miniconda3/envs/assembly/bin/quast.py \
-t 12 --report-all-metrics \
-b --fungus --k-mer-stats \
--nanopore $READ_DIR/SRR30482968.fastq.gz \
--pe1 $READ_DIR/SRR30482970_R1_trim.fq.gz \
--pe2 $READ_DIR/SRR30482970_R2_trim.fq.gz \
--labels "Aaurelia_Canu, Aaurelia_Flye, Aaurelia_NextDenovo, Aaurelia_quickmerge" \
-o ${WORK_DIR}/quast \
$HOME/Aaurelia_canu/Aaurelia_canu.fasta \
$HOME/Aaurelia_flye/assembly.fasta \
$HOME/Aaurelia_nd/03.ctg_graph/nd.asm.fasta \
$HOME/Aaurelia_merged.fasta \
$HOME/Aaurelia_merged.teloclip.fasta
```

#### Assessing assembly completeness using `compleasm`
The completeness of the genome assembly can be assessed by querying the genome for core genes, here core ascomycete genes. We used `compleasm v0.2.6` ([Manual](https://github.com/huangnengCSU/compleasm)):
```ShellSession
# Assembly Arachnopeziza aurata
compleasm run -t 4 -a $HOME/Aaurata_merged.teloclip.fasta \
-l ascomycota_odb10 \
-L $HOME/Database/BUSCO \
-o $WORK_DIR/compleasm/Aaurata_merged

# Assembly Arachnopeziza aurelia
compleasm run -t 4 -a $HOME/Aaurelia_merged.teloclip.fasta \
-l ascomycota_odb10 \
-L $HOME/Database/BUSCO \
-o $WORK_DIR/compleasm/Aaurelia_merged

# Previously published assembly Arachnopeziza araneosa
compleasm run -t 4 -a $HOME/Arachnopeziza_araneosa/GCA_003988855.1_ASM398885v1_genomic.fna \
-l ascomycota_odb10 \
-L $HOME/Database/BUSCO \
-o $WORK_DIR/compleasm/Aaraneosa
```

#### Estimating assembly errors with `CRAQ`
We used `CRAQ v1.0.9` to assess the accuracy of the assembled genomic sequences ([Manual](https://github.com/JiaoLaboratory/CRAQ)). To this end, we first remapped the long reads to the genomes using `Minimap2` and then ran the `CRAQ` analysis:
```ShellSession
# Assembly Arachnopeziza aurata

# Map ONP reads to assembly
minimap2 -t 8 -k 20 -ax map-ont -R "@RG\tID:foo\tSM:bar1\tLB:lib" \
$HOME/Aaurata_merged.teloclip.fasta \
$READ_DIR/SRR30482969.fastq.gz | samtools sort -m 2G -@ 8 -o ${WORK_DIR}/Aaurata_merged.teloclip.minimap.bam
# Index
samtools index -@ 8 ${WORK_DIR}/Aaurata_merged.teloclip.minimap.bam

# Run CRAQ
perl $HOME/utilities/CRAQ/bin/craq -t 8 -pl T \
-g $HOME/Aaurata_merged.teloclip.fasta \
-sms ${WORK_DIR}/Aaurata_merged.teloclip.minimap.bam \
-D ${WORK_DIR}/Aaurata_craq
```

```
# Assembly Arachnopeziza aurelia

# Map ONP reads to assembly
minimap2 -t 8 -k 20 -ax map-ont -R "@RG\tID:foo\tSM:bar1\tLB:lib" \
$HOME/Aaurelia_merged.teloclip.fasta \
$READ_DIR/SRR30482968.fastq.gz | samtools sort -m 2G -@ 8 -o ${WORK_DIR}/Aaurelia_merged.teloclip.minimap.bam
# Index
samtools index -@ 8 ${WORK_DIR}/Aaurelia_merged.teloclip.minimap.bam

# Run CRAQ
perl $HOME/utilities/CRAQ/bin/craq -t 8 -pl T \
-g $HOME/Aaurelia_merged.teloclip.fasta \
-sms ${WORK_DIR}/Aaurelia_merged.teloclip.minimap.bam \
-D ${WORK_DIR}/Aaurelia_craq
```
