#### Preparing datasets for circos plots

Several datasets needed to be prepared for generating meaningful circos plots for the genome assemblies of *Arachnopeziza aurata* and *Arachnopeziza aurelia*. 

##### Cytoband 
Text

##### GC content map
Text

##### TE and gene density map
Text

#### Generating circos plots for the genome assemblies of *Arachnopeziza aurata* and *Arachnopeziza aurelia*

Circos plots were generated with `circlize v0.4.10` ([Manual](https://jokergoo.github.io/circlize_book/book/)) in `R v4.3.1`, using the datasets prepared above. The codes are shown below. 

Plot for *Arachnopeziza aurata*:
```R
# set working directory
setwd("${PATH_TO_DATA}/Circos_plot/")

# call required libraries
library(circlize)
library(extrafont)
font_import(pattern = "arial", prompt = FALSE)
loadfonts(device = "pdf")

# load data
cytoband <- read.table("Aaurata_ChrData_Rcircos.txt", header=T, sep="\t")
gc <- read.table("Aaurata_GC.windows.txt", header=T, sep="\t")
gene_density <- read.table("Aaurata_gene.windows.txt", header=T, sep="\t")
te_density <- read.table("Aaurata_TE.windows.txt", header=T, sep="\t")
annot <- read.table("Aaurata.rRNA.txt", header=T, sep="\t")

# prepare cytoband data
cytoband.df = read.table("Aaurelia_cytoband_Rcircos.txt", 
                         colClasses = c("character", "numeric",
                                        "numeric", "character", "character"), 
                         sep = "\t", header = T)

# Define a function to assign colors based on the gieStain column
cytoband_color <- function(gieStain) {
  if (gieStain == "gneg") return("#FFFFFF")
  else if (gieStain == "gpos50") return("#808080")
  else if (gieStain == "gpos100") return("#000000")
  else if (gieStain == "gvar") return("#CCCCCC")
  else return("#FFDDC1")  # Default color if gieStain is unrecognized
}

# set up PDF
pdf("./Aaurata.circos.pdf", width = 850/72, height = 750/72, onefile = FALSE)

# set up circos
circos.clear()
col_text <- "#999999"
circos.par("track.height" = 0.2, start.degree = 90,
           #gap.after = c(rep(2, 15), 10),
           gap.degree=c(rep(2, 15), 20), cell.padding=c(0, 0, 0, 0))
circos.initialize(factors=cytoband$chr, 
                  xlim=matrix(c(rep(0, 16), cytoband$end), ncol=2)
                  )

# genomes
circos.track(ylim=c(0, 1), panel.fun=function(x, y) {
  chr=CELL_META$sector.index
  xlim=CELL_META$xlim
  ylim=CELL_META$ylim
  circos.text(mean(xlim), mean(ylim), chr, cex=1, col="#000000", 
              facing="bending.outside", niceFacing=TRUE,
              #family = "Arial",
              adj = c(0.5, 4.5))
}, bg.col="#cccccc", bg.border=F, track.height=0.05)

# Add additional colored areas from cytoband.df
cytoband.df$col <- sapply(cytoband.df$gieStain, cytoband_color)

# Adding additional regions to the track
for(i in 1:nrow(cytoband.df)) {
  circos.rect(xleft = cytoband.df$start[i], 
              ybottom = 0, 
              xright = cytoband.df$end[i], 
              ytop = 1, 
              sector.index = cytoband.df$chr[i], 
              col = cytoband.df$col[i], 
              border = NA)
}

# add x-axis
brk <- c(0, 0.5, 1, 1.5, 2, 2.5, 3, 3.5, 4, 4.5, 5)*10^6
circos.track(track.index = get.current.track.index(), panel.fun=function(x, y) {
  circos.axis(h="top", major.at=brk, labels=round(brk/10^6, 1), labels.cex=0.75, 
              col="#999999", labels.col=col_text, lwd=0.7, 
              labels.facing="clockwise")
}, bg.border=F)

# rDNA labels
circos.genomicLabels(annot, labels.column=5, cex=0.75, col=col_text, 
                     line_lwd=0.5, line_col="#999999", 
                     side="inside", connection_height=0.02, labels_height=0.04,
                     track.margin = c(0.02,0))

### tracks
# GC content track
circos.track(factors=gc$chr, x=gc$start, y=gc$value, panel.fun=function(x, y) {
  circos.lines(x, y, col="#999999", lwd=0.6)
  # Get the current chromosome from the sector index
  current_chr <- CELL_META$sector.index
  # Filter the cytoband data for the current chromosome
  chr_end <- cytoband$end[cytoband$chr == current_chr]
  chr_end <- max(chr_end)  # Use the maximum end value
  # line segments
  circos.segments(x0=0, x1=chr_end, y0=0.25, y1=0.25, lwd=0.6, lty="11", col="grey90")
  circos.segments(x0=0, x1=chr_end, y0=0.5, y1=0.5, lwd=0.6, lty="11", col="grey90")
  circos.segments(x0=0, x1=chr_end, y0=0.75, y1=0.75, lwd=0.6, lty="11", col="grey90")
}, ylim=c(0,1), track.height=0.15, bg.border=T)
# GC y axis
circos.yaxis(sector.index = "Chr01", at=c(0.25, 0.5, 0.75), labels.cex=0.75, lwd=0, tick.length=0, labels.col=col_text, col="#FFFFFF")

# Gene density track
circos.track(factors=gene_density$chr, x=gene_density$start, y=gene_density$value, panel.fun=function(x, y) {
  circos.lines(x, y, col="#0072B2", lwd=0.6)
  # Get the current chromosome from the sector index
  current_chr <- CELL_META$sector.index
  # Filter the cytoband data for the current chromosome
  chr_end <- cytoband$end[cytoband$chr == current_chr]
  chr_end <- max(chr_end)  # Use the maximum end value
  # line segments
  circos.segments(x0=0, x1=chr_end, y0=50, y1=50, lwd=0.6, lty="11", col="grey90")
  circos.segments(x0=0, x1=chr_end, y0=100, y1=100, lwd=0.6, lty="11", col="grey90")
  circos.segments(x0=0, x1=chr_end, y0=150, y1=150, lwd=0.6, lty="11", col="grey90")
}, ylim=range(gene_density$value), track.height=0.15, bg.border=T)
# Gene density y axis
circos.yaxis(sector.index = "Chr01", at=c(0, 50, 100, 150), labels.cex=0.75, lwd=0, tick.length=0, labels.col=col_text, col="#FFFFFF")

# TE density track
circos.track(factors=te_density$chr, x=te_density$start, y=te_density$value, panel.fun=function(x, y) {
  circos.lines(x, y, col="#CC79A7", lwd=0.6)
  # Get the current chromosome from the sector index
  current_chr <- CELL_META$sector.index
  # Filter the cytoband data for the current chromosome
  chr_end <- cytoband$end[cytoband$chr == current_chr]
  chr_end <- max(chr_end)  # Use the maximum end value
  # line segments
  circos.segments(x0=0, x1=chr_end, y0=5, y1=5, lwd=0.6, lty="11", col="grey90")
  circos.segments(x0=0, x1=chr_end, y0=10, y1=10, lwd=0.6, lty="11", col="grey90")
  circos.segments(x0=0, x1=chr_end, y0=15, y1=15, lwd=0.6, lty="11", col="grey90")
}, ylim=range(te_density$value), track.height=0.15, bg.border=T)
# TE density y axis
circos.yaxis(sector.index = "Chr01", at=c(0, 5, 10, 15), labels.cex=0.75, lwd=0, tick.length=0, labels.col=col_text, col="#FFFFFF")

# print plot
dev.off()
```

Plot for **Arachnopeziza aurelia*
```R
# set working directory
setwd("${PATH_TO_DATA}/Circos_plot/")

# call required libraries
library(circlize)
library(extrafont)
font_import(pattern = "arial", prompt = FALSE)
loadfonts(device = "pdf")

# load data
cytoband <- read.table("Aaurelia_ChrData_Rcircos.txt", header=T, sep="\t")
gc <- read.table("Aaurelia_GC.windows.txt", header=T, sep="\t")
gene_density <- read.table("Aaurelia_gene.windows.txt", header=T, sep="\t")
te_density <- read.table("Aaurelia_TE.windows.txt", header=T, sep="\t")
annot <- read.table("Aaurelia.rRNA.txt", header=T, sep="\t")

# prepare cytoband data
cytoband.df = read.table("Aaurelia_cytoband_Rcircos.txt", 
                         colClasses = c("character", "numeric",
                                        "numeric", "character", "character"), 
                         sep = "\t", header = T)

# Define a function to assign colors based on the gieStain column
cytoband_color <- function(gieStain) {
  if (gieStain == "gneg") return("#FFFFFF")
  else if (gieStain == "gpos50") return("#808080")
  else if (gieStain == "gpos100") return("#000000")
  else if (gieStain == "gvar") return("#CCCCCC")
  else return("#FFDDC1")  # Default color if gieStain is unrecognized
}

# set up PDF
pdf("./Aaurelia.circos.pdf", width = 850/72, height = 750/72, onefile = FALSE)

# set up circos
circos.clear()
col_text <- "#999999"
circos.par("track.height" = 0.2, start.degree = 90,
           #gap.after = c(rep(2, 15), 10),
           gap.degree=c(rep(2, 15), 20), cell.padding=c(0, 0, 0, 0))
circos.initialize(factors=cytoband$chr, 
                  xlim=matrix(c(rep(0, 16), cytoband$end), ncol=2)
                  )

# genomes
circos.track(ylim=c(0, 1), panel.fun=function(x, y) {
  chr=CELL_META$sector.index
  xlim=CELL_META$xlim
  ylim=CELL_META$ylim
  circos.text(mean(xlim), mean(ylim), chr, cex=1, col="#000000", 
              facing="bending.outside", niceFacing=TRUE,
              #family = "Arial",
              adj = c(0.5, 4.5))
}, bg.col="#cccccc", bg.border=F, track.height=0.05)

# Add additional colored areas from cytoband.df
cytoband.df$col <- sapply(cytoband.df$gieStain, cytoband_color)

# Adding additional regions to the track
for(i in 1:nrow(cytoband.df)) {
  circos.rect(xleft = cytoband.df$start[i], 
              ybottom = 0, 
              xright = cytoband.df$end[i], 
              ytop = 1, 
              sector.index = cytoband.df$chr[i], 
              col = cytoband.df$col[i], 
              border = NA)
}

# add x-axis
brk <- c(0, 0.5, 1, 1.5, 2, 2.5, 3, 3.5, 4, 4.5, 5)*10^6
circos.track(track.index = get.current.track.index(), panel.fun=function(x, y) {
  circos.axis(h="top", major.at=brk, labels=round(brk/10^6, 1), labels.cex=0.75, 
              col="#999999", labels.col=col_text, lwd=0.7, 
              labels.facing="clockwise")
}, bg.border=F)

# rDNA labels
circos.genomicLabels(annot, labels.column=5, cex=0.75, col=col_text, 
                     line_lwd=0.5, line_col="#999999", 
                     side="inside", connection_height=0.02, labels_height=0.04,
                     track.margin = c(0.02,0))

### tracks
# GC content track
circos.track(factors=gc$chr, x=gc$start, y=gc$value, panel.fun=function(x, y) {
  circos.lines(x, y, col="#999999", lwd=0.6)
  # Get the current chromosome from the sector index
  current_chr <- CELL_META$sector.index
  # Filter the cytoband data for the current chromosome
  chr_end <- cytoband$end[cytoband$chr == current_chr]
  chr_end <- max(chr_end)  # Use the maximum end value
  # line segments
  circos.segments(x0=0, x1=chr_end, y0=0.25, y1=0.25, lwd=0.6, lty="11", col="grey90")
  circos.segments(x0=0, x1=chr_end, y0=0.5, y1=0.5, lwd=0.6, lty="11", col="grey90")
  circos.segments(x0=0, x1=chr_end, y0=0.75, y1=0.75, lwd=0.6, lty="11", col="grey90")
}, ylim=c(0,1), track.height=0.15, bg.border=T)
# GC y axis
circos.yaxis(sector.index = "Chr01", at=c(0.25, 0.5, 0.75), labels.cex=0.75, lwd=0, tick.length=0, labels.col=col_text, col="#FFFFFF")

# Gene density track
circos.track(factors=gene_density$chr, x=gene_density$start, y=gene_density$value, panel.fun=function(x, y) {
  circos.lines(x, y, col="#0072B2", lwd=0.6)
  # Get the current chromosome from the sector index
  current_chr <- CELL_META$sector.index
  # Filter the cytoband data for the current chromosome
  chr_end <- cytoband$end[cytoband$chr == current_chr]
  chr_end <- max(chr_end)  # Use the maximum end value
  # line segments
  circos.segments(x0=0, x1=chr_end, y0=50, y1=50, lwd=0.6, lty="11", col="grey90")
  circos.segments(x0=0, x1=chr_end, y0=100, y1=100, lwd=0.6, lty="11", col="grey90")
  circos.segments(x0=0, x1=chr_end, y0=150, y1=150, lwd=0.6, lty="11", col="grey90")
}, ylim=range(gene_density$value), track.height=0.15, bg.border=T)
# Gene density y axis
circos.yaxis(sector.index = "Chr01", at=c(0, 50, 100, 150), labels.cex=0.75, lwd=0, tick.length=0, labels.col=col_text, col="#FFFFFF")

# TE density track
circos.track(factors=te_density$chr, x=te_density$start, y=te_density$value, panel.fun=function(x, y) {
  circos.lines(x, y, col="#CC79A7", lwd=0.6)
  # Get the current chromosome from the sector index
  current_chr <- CELL_META$sector.index
  # Filter the cytoband data for the current chromosome
  chr_end <- cytoband$end[cytoband$chr == current_chr]
  chr_end <- max(chr_end)  # Use the maximum end value
  # line segments
  circos.segments(x0=0, x1=chr_end, y0=5, y1=5, lwd=0.6, lty="11", col="grey90")
  circos.segments(x0=0, x1=chr_end, y0=10, y1=10, lwd=0.6, lty="11", col="grey90")
  circos.segments(x0=0, x1=chr_end, y0=15, y1=15, lwd=0.6, lty="11", col="grey90")
}, ylim=range(te_density$value), track.height=0.15, bg.border=T)
# TE density y axis
circos.yaxis(sector.index = "Chr01", at=c(0, 5, 10, 15), labels.cex=0.75, lwd=0, tick.length=0, labels.col=col_text, col="#FFFFFF")

# print plot
dev.off()
```
