#### Genome synteny with `MUMmer` and `Rideogram`
We compared the genome assemblies of *Arachnopeziza aurata* and *Arachnopeziza aurelia* by collinearity analysis (synteny) using `MUMmer v4.4.0' ([Manual]()). 
```ShellSession
# generating matches with nucmer
nucmer -p ${WORK_DIR}/Aaurata-Aaurelia $HOME/Aaurelia_merged.teloclip.fasta $HOME/Aaurata_merged.teloclip.fasta
dnadiff -d ${WORK_DIR}/Aaurata-Aaurelia.delta -p ${WORK_DIR}/Aaurata-Aaurelia.dnadiff

# filtering alignments and generating dotplot
for infile in ${WORK_DIR}/*delta
do
  j=$(basename $infile .delta)
  delta-filter -l 1000 -q ${infile} > ${j}_filter.delta
  show-coords -c -l -L 1000 -r -T ${j}_filter.delta > ${j}_filter_coords.txt
  mummerplot -f -p ${j}_filter --png ${j}_filter.delta
done
```

We visualized collinearity with `RIdeogram v0.2.2` ([Manual](https://github.com/TickingClock1992/RIdeogram)), as follows:
```R

```


#### Repeat content
First, we created custom repeat libraries using `EDTA2` ([Manual](https://github.com/oushujun/EDTA)) and `TEtrimmer` ([Manual](https://github.com/qjiangzhao/TEtrimmer)). Initial libraries were calculated with `EDTA2` like this:
```ShellSession
# Assembly Arachnopeziza aurata
perl $HOME/EDTA/EDTA.pl \
    --genome $HOME/Aaurata_merged.teloclip.fasta \
    --sensitive 1 --anno 1 --threads 50

# Assembly Arachnopeziza aurelia
perl $HOME/EDTA/EDTA.pl \
    --genome $HOME/Aaurelia_merged.teloclip.fasta \
    --sensitive 1 --anno 1 --threads 50
```

Then, `TEtrimmer` was run using the `EDTA2` libraries.
```ShellSession
# Assembly Arachnopeziza aurata
python ./TEtrimmer.py -i ${WORK_DIR}/Aaurata_merged.teloclip.genomic_intact_and_TElib.fa \
                 -g $HOME/Aaurata_merged.teloclip.fasta \
                 -o ${WORK_DIR}/Aaurata_merged.teloclip.genomic_EDTA \
                 --pfam_dir $HOME/pfam_database \
                 -t 50 --classify_all -ga

# Assembly Arachnopeziza aurelia
python ./TEtrimmer.py -i ${WORK_DIR}/Aaurelia_merged.teloclip.genomic_intact_and_TElib.fa \
                 -g $HOME/Aaurelia_merged.teloclip.fasta \
                 -o ${WORK_DIR}/Aaurelia_merged.teloclip.genomic_EDTA \
                 --pfam_dir $HOME/pfam_database \
                 -t 50 --classify_all -ga
```

We used `calcDivergenceFromAlign.pl` to generate the TE divergence plots.
```ShellSession
# Assembly Arachnopeziza aurata
perl $HOME/miniconda3/envs/TEtrimmer_base/share/RepeatMasker/util/calcDivergenceFromAlign.pl \
-s $HOME/Aaurata_merged.teloclip.fasta.align.txt $HOME/Aaurata_merged.teloclip.fasta.align

# Assembly Arachnopeziza aurelia
perl $HOME/miniconda3/envs/TEtrimmer_base/share/RepeatMasker/util/calcDivergenceFromAlign.pl \
-s $HOME/Aaurelia_merged.teloclip.fasta.align.txt $HOME/Aaurelia_merged.teloclip.fasta.align
```

`RepeatMasker` allows the genome-wide identification of transposable elements from a repeat database, here generated using `TEtrimmer`. It also masks repetitive sequences.
```ShellSession
# Assembly Arachnopeziza aurata
RepeatMasker -pa 8 -s -engine rmblast \
-lib ${WORK_DIR}/Arachnopeziza_TEtrimmer/aurata/EDTA2_output/TEtrimmer_output_20240726_154529/TEtrimmer_consensus_merged.fasta \
-small -gff -a \
-dir ${WORK_DIR}/RepeatMasker \
$HOME/Aaurata_merged.teloclip.fasta

# Assembly Arachnopeziza aurelia
RepeatMasker -pa 8 -s -engine rmblast \
-lib ${WORK_DIR}/Arachnopeziza_TEtrimmer/aurelia/EDTA2_output/TEtrimmer_output_20240726_154529/TEtrimmer_consensus_merged.fasta \
-small -gff -a \
-dir ${WORK_DIR}/RepeatMasker \
$HOME/Aaurelia_merged.teloclip.fasta
```

`RepeatMasker` produces a GFF file of repeat elements, a genome file where repetitive elements are masked, and an output table. 

`RepeatMasker` output for the genome assembly of *Arachnpeziza aurata*:
```
==================================================
file name: Aaurata_merged.teloclip.fasta
sequences:            17
total length:   43156266 bp  (43156266 bp excl N/X-runs)
GC level:         44.49 %
bases masked:    1383848 bp ( 3.21 %)
==================================================
               number of      length   percentage
               elements*    occupied  of sequence
--------------------------------------------------
SINEs:               13         2429 bp    0.01 %
      ALUs            0            0 bp    0.00 %
      MIRs            0            0 bp    0.00 %

LINEs:                0            0 bp    0.00 %
      LINE1           0            0 bp    0.00 %
      LINE2           0            0 bp    0.00 %
      L3/CR1          0            0 bp    0.00 %

LTR elements:      1118      1017668 bp    2.36 %
      ERVL            0            0 bp    0.00 %
      ERVL-MaLRs      0            0 bp    0.00 %
      ERV_classI      0            0 bp    0.00 %
      ERV_classII     0            0 bp    0.00 %

DNA elements:       144        85231 bp    0.20 %
     hAT-Charlie      0            0 bp    0.00 %
     TcMar-Tigger     0            0 bp    0.00 %

Unclassified:       134        31814 bp    0.07 %

Total interspersed repeats:  1137142 bp    2.63 %


Small RNA:            0            0 bp    0.00 %

Satellites:           0            0 bp    0.00 %
Simple repeats:    4855       204309 bp    0.47 %
Low complexity:     934        45945 bp    0.11 %
==================================================

* most repeats fragmented by insertions or deletions
  have been counted as one element


The query species was assumed to be homo
RepeatMasker Combined Database: Dfam_3.0, RepBase-20181026

run with rmblastn version 2.14.1+
The query was compared to classified sequences in ".../TEtrimmer_consensus_merged.fasta"
```

`RepeatMasker` output for the genome assembly of *Arachnpeziza aurelia*:
```
==================================================
file name: Aaurelia_merged.teloclip.fasta
sequences:            17
total length:   46414168 bp  (46414168 bp excl N/X-runs)
GC level:         43.26 %
bases masked:    2220872 bp ( 4.78 %)
==================================================
               number of      length   percentage
               elements*    occupied  of sequence
--------------------------------------------------
SINEs:                0            0 bp    0.00 %
      ALUs            0            0 bp    0.00 %
      MIRs            0            0 bp    0.00 %

LINEs:                0            0 bp    0.00 %
      LINE1           0            0 bp    0.00 %
      LINE2           0            0 bp    0.00 %
      L3/CR1          0            0 bp    0.00 %

LTR elements:       582      1160578 bp    2.50 %
      ERVL            0            0 bp    0.00 %
      ERVL-MaLRs      0            0 bp    0.00 %
      ERV_classI      0            0 bp    0.00 %
      ERV_classII     0            0 bp    0.00 %

DNA elements:        43        46982 bp    0.10 %
     hAT-Charlie      0            0 bp    0.00 %
     TcMar-Tigger     0            0 bp    0.00 %

Unclassified:       534       616151 bp    1.33 %

Total interspersed repeats:  1823711 bp    3.93 %


Small RNA:           29        66681 bp    0.14 %

Satellites:           0            0 bp    0.00 %
Simple repeats:    6826       282507 bp    0.61 %
Low complexity:    1045        51151 bp    0.11 %
==================================================

* most repeats fragmented by insertions or deletions
  have been counted as one element


The query species was assumed to be homo
RepeatMasker Combined Database: Dfam_3.0, RepBase-20181026

run with rmblastn version 2.14.1+
The query was compared to classified sequences in ".../TEtrimmer_consensus_merged.fasta"
```
