#### Analysing and plotting `RIPCAL` analysis results

`RIPCAL v2.0` ([Hane & Oliver 2008](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-478)) was used to generate dinucleotide frequencies for the genomes of *Arachnopeziza aurata*, *Arachnopeziza aurelia*, *Blumeria hordei*, and *Rhynchosporium commune*. We analysed the frequencies and plotted the results as shown below using `R v4.3.1`. 

First, histograms to draw GC content distributions:
```R
# home directory
setwd("$PATH/GC_content")

# libraries
library(ggplot2)
library(ggpubr)
library(dplyr)
library(scales)
library(multcompView)
library(reshape2)
library(grid)
library(gridExtra)
library(cowplot)

###theme
newtheme <- theme_classic() + 
  theme(legend.position = c(0.9,0.9)) +
  theme(legend.title = element_blank()) +
  theme(axis.text.x = element_text(color="black", size = 14, hjust=0.5, vjust=0.5)) +
  theme(axis.text.y = element_text(color="black", size = 14, hjust=1, vjust=0.5)) +
  theme(axis.title.x = element_text(face="bold", color="black", size = 14)) +
  theme(axis.title.y = element_text(face="bold", color="black", size = 14)) +
  theme(axis.line = element_line(size = 1)) +
  theme(axis.ticks = element_line(size = 1)) +
  theme(legend.text = element_text(color="black", size = 14))

# data
aurata <- read.table("windows_gc.Aaurata.bed", header=F, sep="\t")
colnames(aurata) <- c("Chr", "start", "end", "AT", "GC",
                      "A", "C", "G", "T", "N", "other", "length")
aurelia <- read.table("windows_gc.Aaurelia.bed", header=F, sep="\t")
colnames(aurelia) <- c("Chr", "start", "end", "AT", "GC",
                      "A", "C", "G", "T", "N", "other", "length")
Bhordei <- read.table("windows_gc.Bhordei.bed", header=F, sep="\t")
colnames(Bhordei) <- c("Chr", "start", "end", "AT", "GC",
                      "A", "C", "G", "T", "N", "other", "length")
Bcinerea <- read.table("windows_gc.Bcinerea.bed", header=F, sep="\t")
colnames(Bcinerea) <- c("Chr", "start", "end", "AT", "GC",
                       "A", "C", "G", "T", "N", "other", "length")
Rcommune <- read.table("windows_gc.Rcommune.bed", header=F, sep="\t")
colnames(Rcommune) <- c("Chr", "start", "end", "AT", "GC",
                       "A", "C", "G", "T", "N", "other", "length")

# create plots
plot.gc.Rcommune <- ggplot(Rcommune, aes(x = GC)) +
  geom_histogram(fill = "#56B4E9", position = "identity", alpha= 0.7, bins = 60) +
  xlim(0,0.7) + ### set x axis limitation 
  scale_y_continuous(limits = c(-1, 2200), expand = c(0, 0.25)) +
  labs(x = "GC content", 
       y = "Frequency") +
  theme_classic() +
  newtheme
plot.gc.Bhordei <- ggplot(Bhordei, aes(x = GC)) +
  geom_histogram(fill = "#56B4E9", position = "identity", alpha= 0.7, bins = 60) +
  xlim(0,0.7) + ### set x axis limitation 
  scale_y_continuous(limits = c(-1, 4200), expand = c(0, 0.25)) +
  labs(x = "GC content", 
       y = "Frequency") +
  theme_classic() +
  newtheme
plot.gc.Bcinerea <- ggplot(Bcinerea, aes(x = GC)) +
  geom_histogram(fill = "#56B4E9", position = "identity", alpha= 0.7, bins = 60) +
  xlim(0,0.7) + ### set x axis limitation 
  scale_y_continuous(limits = c(-1, 2200), expand = c(0, 0.25)) +
  labs(x = "GC content", 
       y = "Frequency") +
  theme_classic() +
  newtheme
plot.gc.aurata <- ggplot(aurata, aes(x = GC)) +
  geom_histogram(fill = "#56B4E9", position = "identity", alpha= 0.7, bins = 60) +
  xlim(0,0.7) + ### set x axis limitation 
  scale_y_continuous(limits = c(-1, 2200), expand = c(0, 0.25)) +
  labs(x = "GC content", 
       y = "Frequency") +
  theme_classic() +
  newtheme
plot.gc.aurelia <- ggplot(aurelia, aes(x = GC)) +
  geom_histogram(fill = "#56B4E9", position = "identity", alpha= 0.7, bins = 60) +
  xlim(0,0.7) + ### set x axis limitation 
  scale_y_continuous(limits = c(-1, 2200), expand = c(0, 0.25)) +
  labs(x = "GC content", 
       y = "Frequency") +
  theme_classic() +
  newtheme

### combine plots
plot <- plot_grid(plot.gc.aurata, plot.gc.aurelia, plot.gc.Bhordei, plot.gc.Bcinerea, plot.gc.Rcommune,
                  nrow = 1, ncol = 5, align='hv', rel_heights = c(1))
ggsave(filename="../RIP-analysis/GC_histograms.pdf", plot=plot, width = 18, height = 5)
```

Then, we analyse the data generated by `RIPCAL`.
```R
# data
aurata <- read.table("../RIP-analysis/Aaurata_ripcal.txt", header=T, sep="\t")
aurelia <- read.table("../RIP-analysis/Aaurelia_ripcal.txt", header=T, sep="\t")
bhordei <- read.table("../RIP-analysis/Bhordei_ripcal.txt", header=T, sep="\t")
bcinerea <- read.table("../RIP-analysis/Bcinerea_ripcal.txt", header=T, sep="\t")
rcommune <- read.table("../RIP-analysis/Rcommune_ripcal.txt", header=T, sep="\t")

# Calculate CpG rates
aurata.rip <- transform(aurata, 
                        TpA_ApT = ta/at,
                        CpATpG_ApCGpT = (ca + tg) / (ac + gt),
                        CpATpG_TpA = (ca + tg) / ta, 
                        CpCGpG_TpCGpA = (cc + gg) / (tc + ga),
                        CpGTpG_CpA = (cg + tg) / ca,
                        CpTApG_TpTApA = (ct + ag) / (tt + aa))
#
aurelia.rip <- transform(aurelia, 
                        TpA_ApT = ta/at,
                        CpATpG_ApCGpT = (ca + tg) / (ac + gt),
                        CpATpG_TpA = (ca + tg) / ta, 
                        CpCGpG_TpCGpA = (cc + gg) / (tc + ga),
                        CpGTpG_CpA = (cg + tg) / ca,
                        CpTApG_TpTApA = (ct + ag) / (tt + aa))
#
bhordei.rip <- transform(bhordei, 
                         TpA_ApT = ta/at,
                         CpATpG_ApCGpT = (ca + tg) / (ac + gt),
                         CpATpG_TpA = (ca + tg) / ta, 
                         CpCGpG_TpCGpA = (cc + gg) / (tc + ga),
                         CpGTpG_CpA = (cg + tg) / ca,
                         CpTApG_TpTApA = (ct + ag) / (tt + aa))
#
bcinerea.rip <- transform(bcinerea, 
                         TpA_ApT = ta/at,
                         CpATpG_ApCGpT = (ca + tg) / (ac + gt),
                         CpATpG_TpA = (ca + tg) / ta, 
                         CpCGpG_TpCGpA = (cc + gg) / (tc + ga),
                         CpGTpG_CpA = (cg + tg) / ca,
                         CpTApG_TpTApA = (ct + ag) / (tt + aa))
#
rcommune.rip <- transform(rcommune, 
                         TpA_ApT = ta/at,
                         CpATpG_ApCGpT = (ca + tg) / (ac + gt),
                         CpATpG_TpA = (ca + tg) / ta, 
                         CpCGpG_TpCGpA = (cc + gg) / (tc + ga),
                         CpGTpG_CpA = (cg + tg) / ca,
                         CpTApG_TpTApA = (ct + ag) / (tt + aa))
```

Parsing the data from `RIPCAL` for plotting:
```R
# parse dataframes for plotting
df.aurata <- melt(data.frame(aurata.rip[1], aurata.rip[18:23]))
df.aurata$repeatclass <- factor(df.aurata$repeatclass, 
                                 levels = c("Genome", "CDS", "Ty1", "Ty3", "DNA"))
df.aurata$variable <- factor(df.aurata$variable, 
                              levels = c("TpA_ApT", "CpGTpG_CpA", 
                                         "CpATpG_ApCGpT", "CpATpG_TpA", 
                                         "CpTApG_TpTApA", "CpCGpG_TpCGpA"))
#
df.aurelia <- melt(data.frame(aurelia.rip[1], aurelia.rip[18:23]))
df.aurelia$repeatclass <- factor(df.aurelia$repeatclass, 
                                 levels = c("Genome", "CDS", "Ty1", "Ty3", "DNA"))
df.aurelia$variable <- factor(df.aurelia$variable, 
                              levels = c("TpA_ApT", "CpGTpG_CpA", 
                                         "CpATpG_ApCGpT", "CpATpG_TpA", 
                                         "CpTApG_TpTApA", "CpCGpG_TpCGpA"))
#
df.bhordei <- melt(data.frame(bhordei.rip[1], bhordei.rip[18:23]))
df.bhordei$repeatclass <- factor(df.bhordei$repeatclass, 
                                  levels = c("Genome", "CDS", "Ty1", "Ty3", "DNA"))
df.bhordei$variable <- factor(df.bhordei$variable, 
                               levels = c("TpA_ApT", "CpGTpG_CpA", 
                                          "CpATpG_ApCGpT", "CpATpG_TpA", 
                                          "CpTApG_TpTApA", "CpCGpG_TpCGpA"))
#
df.bcinerea <- melt(data.frame(bcinerea.rip[1], bcinerea.rip[18:23]))
df.bcinerea$repeatclass <- factor(df.bcinerea$repeatclass, 
                                  levels = c("Genome", "CDS", "Ty1", "Ty3", "DNA"))
df.bcinerea$variable <- factor(df.bcinerea$variable, 
                               levels = c("TpA_ApT", "CpGTpG_CpA", 
                                          "CpATpG_ApCGpT", "CpATpG_TpA", 
                                          "CpTApG_TpTApA", "CpCGpG_TpCGpA"))
#
df.rcommune <- melt(data.frame(rcommune.rip[1], rcommune.rip[18:23]))
df.rcommune$repeatclass <- factor(df.rcommune$repeatclass, 
                                  levels = c("Genome", "CDS", "Ty1", "Ty3", "DNA"))
df.rcommune$variable <- factor(df.rcommune$variable, 
                               levels = c("TpA_ApT", "CpGTpG_CpA", 
                                          "CpATpG_ApCGpT", "CpATpG_TpA", 
                                          "CpTApG_TpTApA", "CpCGpG_TpCGpA"))
```

Plotting the `RIPCAL` data:
```R
# colors
categories_color <- c(
  'Ty3' = "#009E73",  # Green
  'LINE' = "#D55E00",  # Red-orange
  'Ty1' = "#F0E442", # Yellow
  'DNA' = "#0072B2",  # Blue
  'CDS' = "#c0c0c0",  # Light-grey
  'Genome' = "#999999"  # Grey
)
# x axis labels
xlabs <- c(
  "TpA_ApT" = "TpA/ApT", 
  "CpGTpG_CpA" = "(CpG+TpG)/CpA", 
  "CpATpG_ApCGpT" = "(CpA+TpG)/(ApC+GpT)", 
  "CpATpG_TpA" = "(CpA+TpG)/TpA", 
  "CpTApG_TpTApA" = "(CpT+ApG)/(TpT+ApA)", 
  "CpCGpG_TpCGpA" = "(CpC+GpG)/(TpC+GpA)"
)
# names
categories <- c(
  "Genome" = "Genome", 
  "CDS" = "CDS", 
  "Ty1" = "Ty1/Copia", 
  "Ty3" = "Ty3/mdg4", 
  "DNA" = "DNA transposon"
)

# plots
plot.rip.aurata <- ggplot(df.aurata, 
                          aes(x = variable, y = value, fill = repeatclass)) + 
  geom_bar(stat = "identity", position = "dodge", width = 0.8, col = c('#cccccc')) +
  scale_fill_manual(values = categories_color) +
  scale_x_discrete(labels = xlabs) +
  scale_y_continuous(expand = c(0,0)) +
  labs(y = "Ratio", x = "") +
  theme_classic() +
  newtheme +
  theme(axis.text.x = element_text(size = 14, hjust=1, vjust=1, angle = 45)) + 
  theme(legend.position = "none",
        legend.title = element_blank())
plot.rip.aurelia <- ggplot(df.aurelia, 
                           aes(x = variable, y = value, fill = repeatclass)) + 
  geom_bar(stat = "identity", position = "dodge", width = 0.8, col = c('#cccccc')) +
  scale_fill_manual(values = categories_color) +
  scale_x_discrete(labels = xlabs) +
  scale_y_continuous(expand = c(0,0)) +
  labs(y = "Ratio", x = "") +
  theme_classic() +
  newtheme +
  theme(axis.text.x = element_text(size = 14, hjust=1, vjust=1, angle = 45)) + 
  theme(legend.position = "none",
        legend.title = element_blank())
plot.rip.Bhordei <- ggplot(df.bhordei, 
                           aes(x = variable, y = value, fill = repeatclass)) + 
  geom_bar(stat = "identity", position = "dodge", width = 0.8, col = c('#cccccc')) +
  scale_fill_manual(values = categories_color) +
  scale_x_discrete(labels = xlabs) +
  scale_y_continuous(expand = c(0,0)) +
  labs(y = "Ratio", x = "") +
  theme_classic() +
  newtheme +
  theme(axis.text.x = element_text(size = 14, hjust=1, vjust=1, angle = 45)) + 
  theme(legend.position = "none",
        legend.title = element_blank())
plot.rip.Bcinerea <- ggplot(df.bcinerea, 
                            aes(x = variable, y = value, fill = repeatclass)) + 
  geom_bar(stat = "identity", position = "dodge", width = 0.8, col = c('#cccccc')) +
  scale_fill_manual(values = categories_color) +
  scale_x_discrete(labels = xlabs) +
  scale_y_continuous(expand = c(0,0)) +
  labs(y = "Ratio", x = "") +
  theme_classic() +
  newtheme +
  theme(axis.text.x = element_text(size = 14, hjust=1, vjust=1, angle = 45)) + 
  theme(legend.position = "none",
        legend.title = element_blank())
plot.rip.Rcommune <- ggplot(df.rcommune, 
                            aes(x = variable, y = value, fill = repeatclass)) + 
  geom_bar(stat = "identity", position = "dodge", width = 0.8, col = c('#cccccc')) +
  scale_fill_manual(values = categories_color) +
  scale_x_discrete(labels = xlabs) +
  scale_y_continuous(expand = c(0,0)) +
  labs(y = "Ratio", x = "") +
  theme_classic() +
  newtheme +
  theme(axis.text.x = element_text(size = 14, hjust=1, vjust=1, angle = 45)) + 
  theme(legend.position = "none",
        legend.title = element_blank())

### combine plots
plot <- plot_grid(plot.rip.aurata, plot.rip.aurelia, plot.rip.Bhordei, plot.rip.Bcinerea, plot.rip.Rcommune,
                  nrow = 1, ncol = 5, align='hv', rel_heights = c(1))
ggsave(filename="../RIP-analysis/RIP_profiles.pdf", plot=plot, width = 18, height = 5)
```
