#### Mapping RNA-seq data

We mapped RNA-seq transcriptome data from *Arachnopeziza aurata* and *Arachnopeziza aurelia* cultivated in liquid PDA medium. We used `HISAT2` ([Manual](https://daehwankimlab.github.io/hisat2/manual/)) for this.

```ShellSession
# Arachnopeziza aurata

# create index
hisat2-build $HOME/Aaurata_merged.teloclip.fasta ${WORK_DIR}/index/Aaurata.idx;
# map reads
hisat2 -p 8 --max-intronlen 1000 -k 10 --no-unal -x ${WORK_DIR}/index/Aaurata.idx \
-1 ${WORK_DIR}/SRR30482965_R1_trim.fq.gz -2 ${WORK_DIR}/SRR30482965_R2_trim.fq.gz | \
samtools sort -l 0 -O bam - | samtools view -F 0x04 -b - > ${WORK_DIR}/SRR30482965.aligned.bam;
samtools index -b ${WORK_DIR}/SRR30482965.aligned.bam
#
hisat2 -p 8 --max-intronlen 1000 -k 10 --no-unal -x ${WORK_DIR}/index/Aaurata.idx \
-1 ${WORK_DIR}/SRR30482966_R1_trim.fq.gz -2 ${WORK_DIR}/SRR30482966_R2_trim.fq.gz | \
samtools sort -l 0 -O bam - | samtools view -F 0x04 -b - > ${WORK_DIR}/SRR30482966.aligned.bam;
samtools index -b ${WORK_DIR}/SRR30482966.aligned.bam
#
hisat2 -p 8 --max-intronlen 1000 -k 10 --no-unal -x ${WORK_DIR}/index/Aaurata.idx \
-1 ${WORK_DIR}/SRR30482967_R1_trim.fq.gz -2 ${WORK_DIR}/SRR30482967_R2_trim.fq.gz | \
samtools sort -l 0 -O bam - | samtools view -F 0x04 -b - > ${WORK_DIR}/SRR30482967.aligned.bam;
samtools index -b ${WORK_DIR}/SRR30482967.aligned.bam
```

```ShellSession
# Arachnopeziza aurelia

# create index
hisat2-build $HOME/Aaurelia_merged.teloclip.fasta ${WORK_DIR}/index/Aaurelia.idx;
# map reads
hisat2 -p 8 --max-intronlen 1000 -k 10 --no-unal -x ${WORK_DIR}/index/Aaurelia.idx \
-1 ${WORK_DIR}/SRR30482962_R1_trim.fq.gz -2 ${WORK_DIR}/SRR30482962_R2_trim.fq.gz | \
samtools sort -l 0 -O bam - | samtools view -F 0x04 -b - > ${WORK_DIR}/SRR30482962.aligned.bam;
samtools index -b ${WORK_DIR}/SRR30482962.aligned.bam
#
hisat2 -p 8 --max-intronlen 1000 -k 10 --no-unal -x ${WORK_DIR}/index/Aaurelia.idx \
-1 ${WORK_DIR}/SRR30482963_R1_trim.fq.gz -2 ${WORK_DIR}/SRR30482963_R2_trim.fq.gz | \
samtools sort -l 0 -O bam - | samtools view -F 0x04 -b - > ${WORK_DIR}/SRR30482963.aligned.bam;
samtools index -b ${WORK_DIR}/SRR30482963.aligned.bam
#
hisat2 -p 8 --max-intronlen 1000 -k 10 --no-unal -x ${WORK_DIR}/index/Aaurelia.idx \
-1 ${WORK_DIR}/SRR30482964_R1_trim.fq.gz -2 ${WORK_DIR}/SRR30482964_R2_trim.fq.gz | \
samtools sort -l 0 -O bam - | samtools view -F 0x04 -b - > ${WORK_DIR}/SRR30482964.aligned.bam;
samtools index -b ${WORK_DIR}/SRR30482964.aligned.bam
```


#### Gene prediction with `BRAKER3`

We used `BRAKER3` ([Manual](https://github.com/Gaius-Augustus/BRAKER)) for evidence-based *ab initio* gene annotation in *Arachnopeziza aurata* and *Arachnopeziza aurelia*. We ran `BRAKER3` with a reference gene set `orthodb/Fungi.fa` downloaded from [OrthoDB v11](https://www.orthodb.org/) and RNA-seq data. We ran `BRAKER3` with the singularity build.
```ShellSession
# Arachnopeziza aurata
singularity exec -B $PWD:${WORK_DIR} $BRAKER_SIF/braker3.sif braker.pl \
--threads=8 --fungus --softmasking --crf \
--workingdir=${WORK_DIR} \
--useexisting --species=Sclerotinia_sclerotiorum \
--genome=Aaurata_merged.teloclip.fasta \
--AUGUSTUS_CONFIG_PATH=$HOME/miniconda3/envs/funannotate/config \
--bam=SRR30482965.aligned.bam,SRR30482966.aligned.bam,SRR30482967.aligned.bam \
--prot_seq=$HOME/database/orthodb/Fungi.fa \
--busco_lineage=ascomycota_odb10 --filterOutShort --gff3

# Arachnopeziza aurelia
singularity exec -B $PWD:${WORK_DIR} $BRAKER_SIF/braker3.sif braker.pl \
--threads=8 --fungus --softmasking --crf \
--workingdir=${WORK_DIR} \
--useexisting --species=Sclerotinia_sclerotiorum \
--genome=Aaurelia_merged.teloclip.fasta \
--AUGUSTUS_CONFIG_PATH=$HOME/miniconda3/envs/funannotate/config \
--bam=SRR30482962.aligned.bam,SRR30482963.aligned.bam,SRR30482964.aligned.bam \
--prot_seq=$HOME/database/orthodb/Fungi.fa \
--busco_lineage=ascomycota_odb10 --filterOutShort --gff3
```
