#### Protein domain annotations

Proteion functions can be inferred by searching for protein domains. We used `PFAM v3.4` to search the proteomes of *Arachnopeziza aurata* and *Arachnopeziza aurelia*. The reference database `Pfam-A.hmm` was obtained from [ftp.ebi.ac.uk](https://ftp.ebi.ac.uk/pub/databases/Pfam/current_release/). We further searched for functional domains with `InterProScan v5.73-104.0` ([Manual](https://interproscan-docs.readthedocs.io/en/v5/))

```ShellSession
# running PFAM search with hmmscan
hmmscan --cpu 16 --tblout ${WORK_DIR}/pfam/Aaurata_braker.pfam.txt $HOME/Database/Pfam/Pfam-A.hmm ${WORK_DIR}/Aaurata_braker.protein.fa
hmmscan --cpu 16 --tblout ${WORK_DIR}/pfam/Aaurelia_braker.pfam.txt $HOME/Database/Pfam/Pfam-A.hmm ${WORK_DIR}/Aaurelia_braker.protein.fa

# running InterProScan search
$HOME/utilities/interproscan/interproscan-5.73-104.0/interproscan.sh \
-cpu 8 -dp \
-t p \
-b ${WORK_DIR}/Aaurata \
-i ${WORK_DIR}/Aaurata_braker.protein.fa

$HOME/utilities/interproscan/interproscan-5.73-104.0/interproscan.sh \
-cpu 8 -dp \
-t p \
-b ${WORK_DIR}/Aaurelia \
-i ${WORK_DIR}/Aaurelia_braker.protein.fa
```

#### Transmembrane domain prediction

We used `TMHMM v2.0c` ([Manual](https://vcru.wisc.edu/simonlab/bioinformatics/programs/tmhmm/TMHMM2.0.html)) to predict transmembrane domains in the proteomes of *Arachnopeziza aurata* and *Arachnopeziza aurelia*.

```ShellSession
tmhmm ${WORK_DIR}/Aaurata_braker.protein.fa > ${WORK_DIR}/tmhmm/Aaurata_braker.tmhmm.out
tmhmm ${WORK_DIR}/Aaurelia_braker.protein.fa > ${WORK_DIR}/tmhmm/Aaurelia_braker.tmhmm.out
```

#### Secretion peptide prediction with `SignalP5.0'

We predicted secretion signals in proteins of *Arachnopeziza aurata* and *Arachnopeziza aurelia* using `SignalP5.0`. `TMHMM v2.0c` was used to identify and remove supposed secreted proteins that are transmembrane proteins.

```ShellSession
# predict secretion signals
signalp -org euk -format short -fasta ${WORK_DIR}/Aaurata_braker.protein.fa -mature -prefix ${WORK_DIR}/signalp/Aaurata_braker.SigP5.out;
signalp -org euk -format short -fasta ${WORK_DIR}/Aaurelia_braker.protein.fa -mature -prefix ${WORK_DIR}/signalp/Aaurelia_braker.SigP5.out;

# identify transmembrane proteins
tmhmm ${WORK_DIR}/signalp/Aaurata_braker.SigP5.out_mature.fasta > ${WORK_DIR}/tmhmm/Aaurata_braker.SigP5-tmhmm.out
tmhmm ${WORK_DIR}/signalp/Aaurelia_braker.SigP5.out_mature.fasta > ${WORK_DIR}/tmhmm/Aaurelia_braker.SigP5-tmhmm.out
```

#### Secondary metabolite clusters with `antiSMASH`

Secondary metabolite gene clusters were searched using `antiSMASH v7.1.0` ([Manual](https://github.com/antismash/antismash)).

```ShellSession
antismash -c 8 -t fungi --smcog-trees --output-dir $WD/antismash --output-basename Aaurata_braker $WD/Aaurata_braker.gbk;
antismash -c 8 -t fungi --smcog-trees --output-dir $WD/antismash --output-basename Aaurelia_braker $WD/Aaurelia_braker.gbk
```

I ran the following script to parse the antismash output into a file containing coordinates with secondary gene clusters.
```ShellSession
# Arachnopeziza aurata
for file in Aaurata_Chr*gbk; do
  paste \
  <(grep LOCUS "$file" | awk '{print $2}' | sed 's/\_/\t/g' | sed 's/Aaurata\_\t/Aaurata/g' | sed 's/Aaurata\t/Aaurata_/g' | sed 's/\-/\t/g') \
  <(grep 'gene' "$file" | awk '1; NR = 1 { exit }' | sed 's/[[:space:]]//g' | sed 's/\/gene\=//g' | sed 's/\"//g') \
  <(grep 'product' "$file" | awk '1; NR = 1 { exit }' | sed 's/[[:space:]]//g' | sed 's/\/product\=//g' | sed 's/\"//g') \
  >> Aaurata.antismash.bed;
done

# Arachnopeziza aurelia
for file in Aaurelia_Chr*gbk; do
  paste \
  <(grep LOCUS "$file" | awk '{print $2}' | sed 's/\_/\t/g' | sed 's/Aaurelia\_\t/Aaurelia/g' | sed 's/Aaurelia\t/Aaurelia_/g' | sed 's/\-/\t/g') \
  <(grep 'gene' "$file" | awk '1; NR = 1 { exit }' | sed 's/[[:space:]]//g' | sed 's/\/gene\=//g' | sed 's/\"//g') \
  <(grep 'product' "$file" | awk '1; NR = 1 { exit }' | sed 's/[[:space:]]//g' | sed 's/\/product\=//g' | sed 's/\"//g') \
  >> Aaurelia.antismash.bed;
done
```
